test/extended/image_ecosystem/mongodb_replica_statefulset.go

```diff
@@ -106,9 +106,18 @@ var _ = g.Describe("[Conformance][image_ecosystem][mongodb][Slow] openshift mong
 			g.By("expecting that we can insert a new record on primary node")
 			mongo := dbutil.NewMongoDB(podNames[0])
 			replicaSet := mongo.(exutil.ReplicaSet)
-			out, err := replicaSet.QueryPrimary(oc, `db.test.save({ "status" : "passed" })`)
-			e2e.Logf("save result: %s\n", out)
-			o.Expect(err).ShouldNot(o.HaveOccurred())
+			done := make(chan struct{}, 1)
+			go func() {
+				defer func() { done <- struct{}{} }()
```

Comment:
Is this done to guarantee the empty struct is passed to the `done` channel on completion?

Reply:
yeah, guarantees the select block exits (it would presumably have exited anyway when the timer expires, but this will make it not have to wait for the timeout).  There's some discussion about this in the obsolete commit.